### ISSUE TITLE
Add TypeScript typing directly in the fs-extra package

### ISSUE THREAD

**Issue by ibratoev** [jprichardson/node-fs-extra#247]
The current typescript typings referenced in the README are seriously outdated. The best way to provide TypeScript typings is adding them directly in the package itself.
More info: https://github.com/typings/typings/issues/322 

@jprichardson, I can contribute the typings for the current version if you have no objections.

**Comment by RyanZim** [comment]
@jprichardson Thoughts?

My concern is that the typings will become outdated, and FYI I am not interested in maintaining them.

@ibratoev Have you tried submitting a PR updating the external ones?

**Comment by ibratoev** [comment]
Yeap, I updated the typings in DefinitelyTyped ;) If you are not interested in having them together with your package, feel free to close the issue.
Cheers

**Comment by RyanZim** [comment]
Closing, @jprichardson reopen if you feel differently.

**Comment by donaldpipowitch** [comment]
Would love to see typings directly in the fs-extra package. Maybe this can be reconsidered?

**Comment by RyanZim** [comment]
AFAIK, none of the core team uses typescript. Because of this, I’m afraid the typings wouldn’t be kept up to date any better than they are now. For that reason, my position stands as before.

**Comment by donaldpipowitch** [comment]
Okay, thank you for the update.

**Comment by jprichardson** [comment]
I'm becoming more interested in it, that's for sure. But I still don't use it. @donaldpipowitch why not the definitely-typed definitions, don't those work well?

**Comment by donaldpipowitch** [comment]
They are "okay". I'd give them a 7 out of 10 in DX points for lib consumers :) I just made a PR for them actually and that's why I wrote here. (Including typings would be a 9/10 and writing the whole project would be a 10/10. 😅)

Minor nits for definitely typed:
- PRs for type fixes can be harder (especially for first time contributors) just because the project is _so_ huge.
- It's harder to maintain typings for prereleases and other "non-stable" versions (probably my biggest concern).
- I find it easier to have typings and documentation in _one_ project. (I recently created _two_ PRs for a project - one to fix typings in the defenetely typed project and one to fix the docs for the project itself.) Essentially the typings are often a better (because stricter) documentation for the project itself. (Interestingly many non-TS developers often fear that the "typings wouldn’t be kept up to date", but are less afraid of documentation to become outdated even though they are "often" not _that_ different.)
- Sometimes the defenetely typed typings don't cover the intent of the original author. (I recently saw this in the `chalk` project which started to include own typings.)
- You don't need to remember the additional typings package :)

(As a lib author myself I also like refactoring my own projects more when they are typed. 😁)

Of course there can also be downsides which I only tell you when you ask for them 😅
